nav2_dwb_controller/dwb_controller/include/dwb_controller/dwb_controller.hpp

```diff
@@ -47,6 +51,10 @@ class DwbController : public rclcpp::Node
   std::shared_ptr<rclcpp::Publisher<geometry_msgs::msg::Twist>> vel_pub_;
   tf2_ros::Buffer tfBuffer_;
   tf2_ros::TransformListener tfListener_;
+
+  // Proxy to the World Model
+  nav2_world_model::WorldModelClient world_model_;
+  TravelDirection travel_direction_;
```

Comment:
The `travel_direction_` seems like unnecessary state. The return code for `checkRegion()` isnt used so the function doesn't seem to have much value. Also the DWB planner will handle the potential velocity commands and compute valid ones or come back with a failure code. Not sure what this is accomplishing

Reply:
Yes, the return code of `checkRegion()` is not used and probably won't be used by DWB. I mainly added that to test the client interface and dynamically visualize some region (in the direction the robot is traveling).

![2](https://user-images.githubusercontent.com/39749557/50924657-13238e00-1405-11e9-8df2-3cde17e194e4.gif)

`travel_direction_` helps in specifying if we should look ahead or backwards

![3](https://user-images.githubusercontent.com/39749557/50924836-862d0480-1405-11e9-8888-15835c177940.gif)

I'm ok with either removing entirely this on DWB or adding some parameter for the user to enable or disable. If we keep it, we should consider the overhead it adds.

 @crdelsey Do you have a preference on this?

Reply:
It's a useful feature, I think though it belongs in a higher level location and uses the clear information to do something (i.e. limit velocity, send a stop command, etc, configurably)

Reply:
@orduno  It's misleading in DWB. It suggests the robot is looking at that region, when that is not how DWB works. DWB only looks at the costmap below the generated trajectories, and it looks both behind and in front, and to the side as well potentially.

Reply:
This is a useful thing to setup boundaries for collision (stop) zones and slow down zones, its super common for autonomous fork lifts and tape following robots to utilize shapes like that to check if it needs to stop. It's a higher level thing, maybe belonging to the robot class implementation of whether to use and the configurations of those shape parameters (and whether to use costmap and/or direct laser scans)

Reply:
Ok, I'll remove it from DWB.